util/changes.go

```diff
@@ -0,0 +1,35 @@
+package util
+
+import (
+	"reflect"
+)
+
+func RecursivelySetUndefinedFields(structToSet, structToDefaultFrom reflect.Value) {
+	if structToSet.Kind() == reflect.Ptr {
+		structToSet = structToSet.Elem()
+	}
+	if structToDefaultFrom.Kind() == reflect.Ptr {
+		structToDefaultFrom = structToDefaultFrom.Elem()
+	}
```

Comment:
I ended up removing this block of code and running the tests for this function and for functions that call this function, and they all still passed. It might be worth it to add a test case that specifically addresses this block of code

Reply:
Done. It was used to guard against deeply nested nil values in defaulted from structs.